pre_commit/clientlib.py

```diff
@@ -112,6 +113,32 @@ def validate_manifest_main(argv: Optional[Sequence[str]] = None) -> int:
 META = 'meta'
 
 
+# should inherit from cfgv.Conditional if sha support is dropped
+class CheckBadRev(cfgv.ConditionalOptional):
+    def check(self, dct: Dict[str, Any]) -> None:
+        super().check(dct)
+
+        if self.key in dct.keys():
+            bad_revs = {
+                'stable',
+                'master',
+                'HEAD',
+            }
+            rev = dct[self.key]
+
+            if '.' not in rev and (
+                not re.match(r'^[a-z0-9]+$', rev) or
+                any(bad_rev in rev for bad_rev in bad_revs) or
+                not rev
```

Comment:
the empty string should warn too, it's (accidentally) an alias for `HEAD`

Reply:
isn't `not rev` creating a warning for empty strings?

Reply:
i also added a test case for an empty `rev` string

Reply:
yeah I think you can remove the `or not rev` -- `'.'` is not contained in `''` and `''` does not match your pattern